src/main/java/net/fabricmc/loom/LoomGradleExtension.java

```diff
@@ -428,4 +438,122 @@ public synchronized File getNextMixinMappings() {
 	public List<LoomDecompiler> getDecompilers() {
 		return decompilers;
 	}
+
+	public void runs(Closure<?> conf) {
+		runs.configure(conf);
+	}
+
+	// Workaround for ugly closures in Kotlin DSL buildscripts
+	public void run(String name, Action<RunConfigSettings> settingsAction) {
+		runs.create(name, settingsAction);
+	}
+
+	public NamedDomainObjectContainer<RunConfigSettings> getRuns() {
+		return runs;
+	}
+
+	public static class RunConfigSettings implements Named {
+		private final List<String> vmArgs = new ArrayList<>();
+		private final List<String> programArgs = new ArrayList<>();
+		private String mode;
+		private String name;
+		private Boolean client;
```

Comment:
I think we should keep the doors open for more than just client or server.

Reply:
Actually this only determines whether `-XstartOnFirstThread` should be applied and whether `KnotClient` should be used if the config isn't named `client` or `server` and no main class has been configured in any installer json (`fabric_installer.<x>.json` should be able to provide main classes for custom modes). But it's indeed a bit strange, so I thought to add a default main class property to the settings for when no main class is defined in any installer json, and just add the `-XstartOnFirstThread` on macos when configuring the default `client` and `server` setup